service/lib/agama/software/manager.rb

```diff
@@ -747,6 +750,41 @@ def copy_zypp_to_target
         FileUtils.copy(glob_credentials, target_dir)
       end
 
+      # private class to ensure that cfa reads installed system
+      # YaST target file does not work reliably as Agama does not have
+      # always switched SCR
+      class TargetFile
+        # Reads file content with respect of changed root in installation.
+        def self.read(path)
+          ::File.read(final_path(path))
+        end
+
+        # Writes file content with respect of changed root in installation.
+        def self.write(path, content)
+          ::File.write(final_path(path), content)
+        end
+
+        def self.final_path(path)
+          ::File.join("/mnt", path)
```

Comment:
```suggestion
          ::File.join(Yast::Installation.destdir, path)
```

This is what the rest of Agama uses. Please add the necessary imports.